didman/api/v1/api_test.go

```diff
@@ -34,462 +32,395 @@ import (
 		assert.ErrorIs(t, err, core.InvalidInputError(""))
+		assert.Nil(t, response)
 	})
 
 	t.Run("error - incorrect did", func(t *testing.T) {
-		ctx := newMockContext(t)
-		ctx.echo.EXPECT().Bind(gomock.Any()).DoAndReturn(func(f interface{}) error {
-			p := f.(*CompoundServiceProperties)
-			*p = request
-			return nil
-		})
-		err := ctx.wrapper.AddCompoundService(ctx.echo, "")
+		test := newMockContext(t)
+		response, err := test.wrapper.AddCompoundService(ctx, AddCompoundServiceRequestObject{Did: "not a did"})
 
 		assert.ErrorIs(t, err, did.ErrInvalidDID)
-		assert.Equal(t, http.StatusBadRequest, ctx.wrapper.ResolveStatusCode(err))
+		assert.Equal(t, http.StatusBadRequest, test.wrapper.ResolveStatusCode(err))
+		assert.Nil(t, response)
 	})
 
 	t.Run("error - incorrect type", func(t *testing.T) {
-		ctx := newMockContext(t)
-		request := CompoundServiceProperties{
-			ServiceEndpoint: serviceEndpoint,
-			Type:            "",
-		}
-		ctx.echo.EXPECT().Bind(gomock.Any()).DoAndReturn(func(f interface{}) error {
-			p := f.(*CompoundServiceProperties)
-			*p = request
-			return nil
+		test := newMockContext(t)
+
+		response, err := test.wrapper.AddCompoundService(ctx, AddCompoundServiceRequestObject{
+			Did: targetDID.String(),
+			Body: &CompoundServiceProperties{
+				Type: "",
+			},
 		})
-		err := ctx.wrapper.AddCompoundService(ctx.echo, id)
 
 		assert.Equal(t, err, core.InvalidInputError("invalid value for type"))
+		assert.Nil(t, response)
 	})
 
 	t.Run("error - invalid service", func(t *testing.T) {
-		ctx := newMockContext(t)
-		ctx.echo.EXPECT().Bind(gomock.Any()).DoAndReturn(func(f interface{}) error {
-			p := f.(*CompoundServiceProperties)
-			*p = request
-			return nil
-		})
-		ctx.didman.EXPECT().AddCompoundService(audit.ContextWithAuditInfo(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, vdr.InvalidServiceError{errors.New("custom error")})
+		test := newMockContext(t)
+		test.didman.EXPECT().AddCompoundService(audit.ContextWithAuditInfo(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, vdr.InvalidServiceError{errors.New("custom error")})
 
-		err := ctx.wrapper.AddCompoundService(ctx.echo, id)
+		response, err := test.wrapper.AddCompoundService(ctx, request)
 
 		assert.ErrorAs(t, err, new(vdr.InvalidServiceError))
-		assert.Equal(t, http.StatusBadRequest, ctx.wrapper.ResolveStatusCode(err))
-	})
-
-	t.Run("error - incorrect post body", func(t *testing.T) {
-		ctx := newMockContext(t)
-		ctx.echo.EXPECT().Bind(gomock.Any()).Return(errors.New("b00m!"))
-
-		err := ctx.wrapper.AddCompoundService(ctx.echo, id)
-
-		assert.EqualError(t, err, "failed to parse v1.CompoundServiceProperties: b00m!")
+		assert.Equal(t, http.StatusBadRequest, test.wrapper.ResolveStatusCode(err))
+		assert.Nil(t, response)
 	})
 }
 
 func TestWrapper_GetCompoundServices(t *testing.T) {
-	idStr := "did:nuts:1#1"
-	id, _ := did.ParseDIDURL(idStr)
+	targetDID := did.MustParseDID("did:nuts:1")
```

Comment:
test has changed by removing the fragment

Reply:
the GetCompoundServices endpoint parses it as `id, err := did.ParseDIDURL(request.Did)` though. Is that wrong too then?

Reply:
fixed